src/rgw/rgw_service.cc

```diff
@@ -272,6 +272,7 @@ void RGWServices_Def::shutdown()
     return;
   }
 
+  datalog_rados.reset();
```

Comment:
why do we have some services that are not being freed or shutdown here?

Reply:
> they probably should all be shutdown here

so, do you want to make this change as part of this PR?

Reply:
i added a commit that calls shutdown on the rest. i have no idea if i got the order right, i just copied the destruction order for the members of `RGWServices_Def`